packages/phone-number-privacy/signer/src/database/wrappers/domainState.ts

```diff
@@ -4,163 +4,133 @@ import Logger from 'bunyan'
 import { Transaction } from 'knex'
 import { Counters, Histograms, Labels } from '../../common/metrics'
 import { getDatabase } from '../database'
-import {
-  DOMAINS_STATES_COLUMNS,
-  DOMAINS_STATES_TABLE,
-  DomainStateRecord,
-} from '../models/domainState'
+import { DomainStateRecord, DOMAIN_STATE_COLUMNS, DOMAIN_STATE_TABLE } from '../models/domainState'
 
-function domainsStates() {
-  return getDatabase()<DomainStateRecord>(DOMAINS_STATES_TABLE)
+function domainStates<D extends Domain>() {
+  return getDatabase()<DomainStateRecord<D>>(DOMAIN_STATE_TABLE)
 }
```

Comment:
Nit: I would argue this method is not actually generic because the return value of this function does not depend on the type parameter (nor can it as a rule in TypeScript). If we have multiple domain types in the future, this database will either hold the state for all domain types (which means the columns used will depend on the type) or we will have multiple databases, which would be a major refactor of this code.
```suggestion
function domainStates() {
  return getDatabase()<DomainStateRecord<Domain>>(DOMAIN_STATE_TABLE)
}
```